pkg/plugins/runtime/k8s/webhooks/injector/probes.go

```diff
@@ -86,12 +86,9 @@ func setVirtualProbesEnabledAnnotation(annotations metadata.Annotations, pod *ku
 	if err != nil {
 		return err
 	}
-	gwEnabled, _, err := metadata.Annotations(pod.Annotations).GetEnabled(metadata.KumaGatewayAnnotation)
-	if err != nil {
-		return err
-	}
 
-	if gwEnabled {
+	gwAnnotation, _ := metadata.Annotations(pod.Annotations).GetString(metadata.KumaGatewayAnnotation)
+	if metadata.IsEnabled(gwAnnotation) {
```

Comment:
I think that the original code is still correct here. In this particular context, the gateway annotation should only be "enabled" or "disabled", so it seems to be as though it's OK to propagate the error.

Reply:
I changed this because otherwise an error is thrown when the annotation is `"builtin"`. Probably the right way to do this is to "verify" the annotation against an allowed set of values in a separate, explicit step.

Reply:
I was expecting that we would not go down this path in the builtin gateway case? Should we force-disable the injector when we see a builtin gateway pod?

Reply:
No, I was testing with a pod and forgot to set SidecarInjection disabled.